### PR TITLE
Refactor error logs in adapter

### PR DESCRIPTION
**Pull request by VajiraPrabuddhaka**
### Purpose
This PR will refactor some error log statements in adapter, according to the formal logging introduced in https://github.com/wso2/product-microgateway/issues/2422

### Issues

Fixes #

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/adapter/adapter.go [hunk 1]
```diff
@@ -317,17 +338,30 @@ func fetchAPIsOnStartUp(conf *config.Config, apiUUIDList []string) {
 			logger.LoggerMgw.Debug("Pushing data to router and enforcer")
 			err := synchronizer.PushAPIProjects(data.Resp, envs)
 			if err != nil {
-				logger.LoggerMgw.Errorf("Error occurred while pushing API data: %v ", err)
+				logger.LoggerMgw.ErrorC(logging.ErrorDetails{
+					Message:   fmt.Sprintf("Error occurred while pushing API data: %v ", err.Error()),
```
Comment (suksw) [inline adapter/internal/adapter/adapter.go:342]:
```suggestion
					Message:   fmt.Sprintf("Error occurred while pushing API data to router and enforcer: %v ", err.Error()),
```

File: adapter/internal/adapter/adapter.go [hunk 2]
```diff
@@ -317,17 +338,30 @@ func fetchAPIsOnStartUp(conf *config.Config, apiUUIDList []string) {
...
+				})
 			}
 			health.SetControlPlaneRestAPIStatus(err == nil)
 		} else if data.ErrorCode >= 400 && data.ErrorCode < 500 {
+			logger.LoggerMgw.ErrorC(logging.ErrorDetails{
+				Message:   fmt.Sprintf("Error occurred when retrieving APIs from control plane(unrecoverable error): %v", data.Err.Error()),
+				Severity:  logging.CRITICAL,
+				ErrorCode: 1106,
+			})
 			logger.LoggerMgw.Errorf("Error occurred when retrieving APIs from control plane: %v", data.Err)
```
Comment (suksw) [inline adapter/internal/adapter/adapter.go:354]: Can we remove this?
```suggestion
```

File: adapter/internal/adapter/adapter.go [hunk 3]
```diff
@@ -117,7 +118,11 @@ func runManagementServer(conf *config.Config, server xdsv3.Server, enforcerServe
 
 	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
 	if err != nil {
-		logger.LoggerMgw.Fatal("failed to listen: ", err)
+		logger.LoggerMgw.ErrorC(logging.ErrorDetails{
+			Message:   fmt.Sprintf("Failed to listen: %v", err.Error()),
```
Comment (slahirucd7) [inline adapter/internal/adapter/adapter.go:122]: Can we include a port number here ? `Failed to listen on port: <portNumber> , error : ....`

**Review by pubudu538 — APPROVED** [review]

File: adapter/internal/api/restserver/config_restapi.go [hunk 4]
```diff
@@ -86,7 +88,11 @@ func configureAPI(api *operations.RestapiAPI) http.Handler {
 	api.BearerTokenAuth = func(token string, scopes []string) (*models.Principal, error) {
 		valid, err := auth.ValidateToken(token, scopes, mgwConfig)
 		if err != nil {
-			logger.LoggerAPI.Error(err.Error())
+			logger.LoggerAPI.ErrorC(logging.ErrorDetails{
+				Message:   fmt.Sprintf("Error occurect while reading the token %v", err.Error()),
```
[the remark this answers is not in the document]
Comment (VajiraPrabuddhaka) [reply adapter/internal/api/restserver/config_restapi.go]: fixed in https://github.com/wso2/product-microgateway/pull/2661/commits/19c337629499420a856f33b0453fda379f1022bf

File: adapter/internal/adapter/adapter.go (hunk 3 above)
Comment (VajiraPrabuddhaka) [reply adapter/internal/adapter/adapter.go]: fixed in https://github.com/wso2/product-microgateway/pull/2661/commits/19c337629499420a856f33b0453fda379f1022bf

File: adapter/internal/auth/auth.go [hunk 5]
```diff
@@ -117,7 +119,11 @@ func GenerateToken(username string) (accessToken string, err error) {
 
 	payload, err = jwt.Sign(token, jwa.RS256, privateKey)
 	if err != nil {
-		loggers.LoggerAuth.Errorf("failed to generate signed payload: %s", err)
+		loggers.LoggerAuth.ErrorC(logging.ErrorDetails{
+			Message:   fmt.Sprintf("failed to generate signed payload: %s", err.Error()),
```
[the remark this answers is not in the document]
Comment (VajiraPrabuddhaka) [reply adapter/internal/auth/auth.go]: Fixed in https://github.com/wso2/product-microgateway/pull/2661/commits/19c337629499420a856f33b0453fda379f1022bf

File: adapter/internal/discovery/xds/consul.go [hunk 6]
```diff
@@ -47,7 +49,11 @@ func startConsulServiceDiscovery(organizationID string) {
 				})
 				query, errConSyn := svcdiscovery.ParseQueryString(consulSyntax)
 				if errConSyn != nil {
-					logger.LoggerXds.Error("consul syntax parse error ", errConSyn)
+					logger.LoggerXds.ErrorC(logging.ErrorDetails{
+						Message:   fmt.Sprintf("consul syntax parse error %v", errConSyn.Error()),
```
[the remark this answers is not in the document]
Comment (VajiraPrabuddhaka) [reply adapter/internal/discovery/xds/consul.go]: Fixed in https://github.com/wso2/product-microgateway/pull/2661/commits/19c337629499420a856f33b0453fda379f1022bf

**Pull request merged**